client/dashboard/sites/trial-ended/index.tsx

```diff
@@ -0,0 +1,124 @@
+// See packages/calypso-products/src/plans-list.tsx.
+const getProduct = ( site: Site ) => {
+	if ( wasEcommerceTrial( site ) ) {
+		return {
+			name: __( 'Commerce' ),
+			tagline: __( 'Grow your online store with commerce-optimized extensions.' ),
+			slug: 'ecommerce-bundle',
+			pathSlug: 'ecommerce',
+		};
+	}
+
+	return {
+		name: __( 'Business' ),
+		tagline: __( 'Unlock next-level WordPress with custom plugins and themes.' ),
+		slug: 'business-bundle',
+		pathSlug: 'business',
+	};
+};
+
+const SiteTrialEnded = ( { siteSlug }: { siteSlug: string } ) => {
+	const [ isSiteDeleteModalOpen, setIsSiteDeleteModalOpen ] = useState( false );
+	const { data: site } = useSuspenseQuery( siteBySlugQuery( siteSlug ) );
+	const product = getProduct( site );
+	const { data: plan } = useQuery( sitePlanBySlugQuery( site.ID, product.slug ) );
+
+	if ( ! plan ) {
+		return null;
+	}
+
+	return (
+		<PageLayout
+			header={
+				<PageHeader
+					title={ __( 'Your free trial has ended' ) }
+					description={ __(
+						'Don’t lose all that hard work! Upgrade to a paid plan to continue working on your site.'
+					) }
+				/>
+			}
+			size="small"
+		>
+			<VStack spacing={ 4 }>
+				<Card>
+					<CardBody>
+						<VStack>
+							<HStack alignment="flex-start">
+								<VStack spacing={ 1 } style={ { flex: 1 } }>
+									<h2 className="site-trial-ended__title">{ product.name }</h2>
+									<Text variant="muted">{ product.tagline }</Text>
+								</VStack>
+								<VStack spacing={ 1 } alignment="flex-end" style={ { flex: 1 } }>
+									<Suspense fallback={ null }>
+										<PlanPrice
+											className="site-trial-ended__price"
+											rawPrice={ ( plan.raw_price_integer * 100 ) / 12 / 100 }
+											currencyCode={ plan.currency_code }
+											isSmallestUnit
+										/>
+									</Suspense>
+									<Text className="site-trial-ended__price-description" variant="muted">
+										{ sprintf(
+											/* translators: %(price)s is the price of the plan */
+											__( 'per month, %(price)s billed annually, excl. taxes' ),
+											{
+												price: plan.formatted_price,
+											}
+										) }
+									</Text>
+								</VStack>
+							</HStack>
+							<HStack>
+								<Button variant="primary" href={ `/checkout/${ site.slug }/${ product.pathSlug }` }>
+									{ __( 'Purchase plan' ) }
+								</Button>
+							</HStack>
+						</VStack>
+					</CardBody>
+				</Card>
+				<HStack spacing={ 3 } justify="flex-start">
+					<Text variant="muted">{ __( 'Not ready to upgrade?' ) }</Text>
+					<ExternalLink href={ `/export/${ site.slug }` }>
+						{ __( 'Export your content' ) }
+					</ExternalLink>
+					<Button variant="link" isDestructive onClick={ () => setIsSiteDeleteModalOpen( true ) }>
+						{ __( 'Delete your site permanently' ) }
+					</Button>
+				</HStack>
+				{ isSiteDeleteModalOpen && (
+					<SiteDeleteModal site={ site } onClose={ () => setIsSiteDeleteModalOpen( false ) } />
```

Comment:
The `<SiteDeleteModal>` feels kinda weird in this situation since you know you already have a plan, that's the message we just read. So it is strange to then get another warning about having an existing plan.

<img width="1112" height="510" alt="CleanShot 2025-08-04 at 10 22 36@2x" src="https://github.com/user-attachments/assets/8e666821-a015-4ed5-b65c-46d8e153ee8d" />

However this is no different from the current v1 behaviour, so not really a blocker.

Reply:
It's strange; the modal works perfectly in my case

<img width="816" height="653" alt="image" src="https://github.com/user-attachments/assets/4456a1c2-d3dc-46c1-802e-7529baffca86" />